File: src/main/java/company/vk/polis/task1/State.java
package company.vk.polis.task1;

public enum State {
    READ,
    UNREAD,
    DELETED;
    private Integer id_of_deleter;
    public Integer getId_of_deleter() { return id_of_deleter; }
    public void setId_of_deleter(Integer id_of_deleter) { this.id_of_deleter = id_of_deleter; }
}

File: src/main/java/company/vk/polis/task1/State.kt
package company.vk.polis.task1

data class State(private val state: StateEnum) {
    private var deletedBy: Long? = null

    constructor(state: StateEnum, deletedBy: Long?) : this(state) {
        if (deletedBy != null && state != StateEnum.DELETED) {
            throw IllegalArgumentException("State can't be with deleted id")
        }
        this.deletedBy = deletedBy
    }

    override fun toString(): String {
        return state.toString()
    }

    fun getDeletedBy(): Long? = deletedBy
    fun getState(): StateEnum = state
}

enum class StateEnum {
    READ, UNREAD, DELETED
}